core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1934,8 +1955,7 @@ class KafkaApis(val requestChannel: RequestChannel,
     val error = resource.`type` match {
       case RResourceType.BROKER => Errors.CLUSTER_AUTHORIZATION_FAILED
       case RResourceType.TOPIC =>
-        // Don't leak topic name unless the user has describe topic permission
-        if (authorize(session, Describe, new Resource(Topic, resource.name)))
+        if (!authorize(session, Describe, new Resource(Topic, resource.name)) || metadataCache.contains(resource.name))
```

Comment:
Hmm.. As far as I can tell, this method is used only after failing DescribeConfigs or AlterConfigs authorization. With this patch, shouldn't we just blindly return `TOPIC_AUTHORIZATION_FAILED`.